CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/CoreGlobalParameters.java

```diff
@@ -51,7 +55,26 @@
      */
     private static final int DATETIME_RANGE_PARAMETER_ID_INDEX = 1;
     public static final String DATETIME_RANGE_PARAMETER_ID = PluginParameter.buildId(CoreGlobalParameters.class, "datetime_range");
-
+    
+    // Map of string types, datetime types, etc.
+    private static final Map<String,PluginParameter> SINGLETON_MAP = new HashMap<>();
+    
+    public static PluginParameter getStringParameterType(final String PARAMETER_ID) {
+        if (SINGLETON_MAP.get(PARAMETER_ID) == null) {
```

Comment:
If checking for whether a key exists, better using `containsKey()`. 
i.e. `if (!SINGLETON_MAP.containsKey(PARAMETER_ID)) {`